test/testdata/parser/error_recovery/def_missing_end_1.rb

```diff
@@ -0,0 +1,12 @@
+# typed: false
+
+class A
+  def test1
+    if x.f
+    end
+
+  def test2
+    if x.f
+    end
+  end
+end # error: unexpected token "end of file"
```

Comment:
> My understanding is that something like this should be addressable within the framework we have here. Is that not the case, or does this test need a TODO?

Yep! I just wanted to break up the change. I will be fixing this in future changes.

> My question is motivated by my surprise that this test and `def_missing_end_2.rb` produce nothing in the class definition. But perhaps that's because the exp files for them are desugar tests, and they should be some kind of parse-tree exp tests?

I've been using desugar-tree tests simply because they're easier to skim to verify correctness. None of the parser formats are as easy to skim. The empty class definition you see is for `<root>` not `A`, which desugar wraps around all parse results, even empty ones.